crates/p2p/src/swarm/setup/events.rs

```diff
@@ -25,6 +28,15 @@ pub enum SetupBehaviourEvent {
         app_public_key: PublicKey,
     },
 
+    /// Emitted when the setup protocol negotiation fails with a remote peer.
+    ///
+    /// This event indicates that the peers could not agree on a common
+    /// version of the setup protocol or that we support Setup, but remote peer does not support
+    /// Setup.
+    ///
+    /// At the moment of writing(end of Aug 2025) we just disconnect from the peer.
+    NegotiationFailed { transport_id: PeerId },
```

Comment:
I think we should also add `ConnectionId` here.
It is available in the `on_connection_handler_event` function that can yield this error.